cpp/benchmarks/search/search.cpp

```diff
@@ -24,50 +24,74 @@
 {
   auto const column_size{static_cast<cudf::size_type>(state.range(0))};
   auto const values_size = column_size;
 
-  auto init_data  = cudf::make_fixed_width_scalar<float>(static_cast<float>(0));
-  auto init_value = cudf::make_fixed_width_scalar<float>(static_cast<float>(values_size));
-  auto step       = cudf::make_fixed_width_scalar<float>(static_cast<float>(-1));
-  auto column     = cudf::sequence(column_size, *init_data);
-  auto values     = cudf::sequence(values_size, *init_value, *step);
+  auto const init_data = cudf::make_fixed_width_scalar<float>(static_cast<float>(0));
+  auto const column    = cudf::sequence(column_size, *init_data);
   if (nulls) {
     auto [column_null_mask, column_null_count] = create_random_null_mask(column->size(), 0.1, 1);
     column->set_null_mask(std::move(column_null_mask), column_null_count);
-    auto [values_null_mask, values_null_count] = create_random_null_mask(values->size(), 0.1, 2);
-    values->set_null_mask(std::move(values_null_mask), values_null_count);
   }
+  auto const values = create_column_data<float>(values_size, nulls);
 
-  auto data_table = cudf::sort(cudf::table_view({*column}));
+  // column is already sorted.
+  auto const data_table = cudf::table_view({*column});
 
-  for (auto _ : state) {
-    cuda_event_timer timer(state, true);
-    auto col = cudf::upper_bound(data_table->view(),
-                                 cudf::table_view({*values}),
-                                 {cudf::order::ASCENDING},
-                                 {cudf::null_order::BEFORE});
+  for ([[maybe_unused]] auto _ : state) {
+    [[maybe_unused]] auto const timer = cuda_event_timer(state, true);
+    [[maybe_unused]] auto const col   = cudf::upper_bound(data_table,
+                                                        cudf::table_view({*values}),
+                                                        {cudf::order::ASCENDING},
+                                                        {cudf::null_order::BEFORE});
   }
 }
 
-BENCHMARK_DEFINE_F(Search, Column_AllValid)(::benchmark::State& state) { BM_column(state, false); }
-BENCHMARK_DEFINE_F(Search, Column_Nulls)(::benchmark::State& state) { BM_column(state, true); }
+#define BINARY_SEARCH_BENCHMARK_DEFINE(name, nulls)   \
+  BENCHMARK_DEFINE_F(BinarySearch, name)              \
+  (::benchmark::State & st) { BM_column(st, nulls); } \
+  BENCHMARK_REGISTER_F(BinarySearch, name)            \
+    ->UseManualTime()                                 \
+    ->Unit(benchmark::kMillisecond)                   \
+    ->Ranges({{100000, 100000000}});
 
-BENCHMARK_REGISTER_F(Search, Column_AllValid)
-  ->UseManualTime()
-  ->Unit(benchmark::kMillisecond)
-  ->Arg(100000000);
-
-BENCHMARK_REGISTER_F(Search, Column_Nulls)
-  ->UseManualTime()
-  ->Unit(benchmark::kMillisecond)
-  ->Arg(100000000);
+BINARY_SEARCH_BENCHMARK_DEFINE(Column_AllValid, false)
+BINARY_SEARCH_BENCHMARK_DEFINE(Column_HasNulls, true)
```

Comment:
Not sure if intentional, but AFAICT we don't need separate definitions here. Validity can be another parameter, something like (if `CreateRange` is available in the version that we use)
```
    ->ArgsProduct({
      benchmark::CreateRange(100000, 100000000, 10),
      {0,1}
    })
```

Reply:
Oh I didn't know that there is such way. I'm trying that...

Reply:
The current Google benchmark version used in cudf doesn't support `CreateRange`. I've created another PR to update it: https://github.com/rapidsai/cudf/pull/11209

Reply:
@vuule It seems that we can't upgrade Google benchmark, so unfortunately your suggestion here can't be worked on.

Reply:
I think we should look at this as an opportunity to convert this benchmark to nvbench. The reason not to allow something like #11209 is to encourage us to switch (which is also important for adjacent initiatives like building dashboards around better benchmark tracking).